Philips.CodeAnalysis.Common/SyntaxNodeSingleDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,87 @@
+﻿// © 2023 Koninklijke Philips N.V. See License.md in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Philips.CodeAnalysis.Common
+{
+	public abstract class SingleDiagnosticAnalyzer<T> : SingleDiagnosticAnalyzer where T : SyntaxNode
+	{
+		protected SyntaxNodeAnalysisContext Context { get; private set; }
+		protected T Node { get; private set; }
+		protected string FullyQualifiedMetaDataName { get; set; }
+
+		protected SingleDiagnosticAnalyzer(DiagnosticId id, string title, string messageFormat, string description, string category,
+											Helper helper = null, DiagnosticSeverity severity = DiagnosticSeverity.Error, bool isEnabled = true)
+			: base(id, title, messageFormat, description, category, helper, severity, isEnabled)
+		{ }
+
+		public override void Initialize(AnalysisContext context)
+		{
+			context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+			context.EnableConcurrentExecution();
+
+			SyntaxKind syntaxKind = GetSyntaxKind();
+
+			if (string.IsNullOrEmpty(FullyQualifiedMetaDataName))
+			{
+				context.RegisterSyntaxNodeAction(StartAnalysis, syntaxKind);
+			}
+			else
+			{
+				context.RegisterCompilationStartAction(startContext =>
+				{
+					if (startContext.Compilation.GetTypeByMetadataName(FullyQualifiedMetaDataName) == null)
+					{
+						return;
```

Comment:
Make this error situation known, maybe an `Exception` even

Reply:
This is not an error situation, but a "feature", taken from the old WinForms analyzer. It allows the analyzer to gracefully short-circuit itself (for performance reasons) if the pre-requisites Reference isn't part of the compilation unit.